Auto update failing for libtbb

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the libtbb package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="bash">
INFO: Updating libtbb [Current version: 2021.10.0]
INFO: package being updated to 2021.11.0.

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 1427k    0 1427k    0     0  3701k      0 --:--:-- --:--:-- --:--:-- 3701k
100 2552k    0 2552k    0     0  4712k      0 --:--:-- --:--:-- --:--:-- 7253k

You are about to commit these changes:

--------------------
diff --git a/packages/libtbb/build.sh b/packages/libtbb/build.sh
index bf1c3677d..ddddcb8c7 100644
--- a/packages/libtbb/build.sh
+++ b/packages/libtbb/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://oneapi-src.github.io/oneTBB/
 TERMUX_PKG_DESCRIPTION="oneAPI Threading Building Blocks"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2021.10.0"
+TERMUX_PKG_VERSION="2021.11.0"
 TERMUX_PKG_SRCURL=https://github.com/oneapi-src/oneTBB/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=487023a955e5a3cc6d3a0d5f89179f9b6c0ae7222613a7185b0227ba0c83700b
+TERMUX_PKG_SHA256=782ce0cab62df9ea125cdea253a50534862b563f1d85d4cda7ad4e77550ac363
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libc++"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
--------------------

bump(main/libtbb): 2021.11.0


Not committing to Git!

INFO: Trying to build package.
INFO: Package libtbb exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building libtbb for arch aarch64...
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13134  100 13134    0     0  35481      0 --:--:-- --:--:-- --:--:-- 35497
Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2253      0 --:--:-- --:--:-- --:--:--  2251
gpg: Signature made Fri 17 Nov 2023 12:25:11 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13320  100 13320    0     0  32962      0 --:--:-- --:--:-- --:--:-- 32970
Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   1983      0 --:--:-- --:--:-- --:--:--  1983
gpg: Signature made Fri 17 Nov 2023 12:26:09 PM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13120  100 13120    0     0  36729      0 --:--:-- --:--:-- --:--:-- 36750
Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   833  100   833    0     0   2330      0 --:--:-- --:--:-- --:--:--  2333
gpg: Signature made Thu 16 Nov 2023 11:44:11 AM UTC
gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
Downloading dependency libc++@26b if necessary...
Found libc++ in https://packages-cf.termux.dev/apt/termux-main/dists/stable
Downloading https://packages-cf.termux.dev/apt/termux-main/pool/main/libc/libc++/libc++_26b_aarch64.deb
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 10  297k   10 30688    0     0  53840      0  0:00:05 --:--:--  0:00:05 53838
100  297k  100  297k    0     0   398k      0 --:--:-- --:--:-- --:--:--  397k
extracting libc++ to /home/builder/.termux-build/_cache-aarch64...
Downloading https://github.com/oneapi-src/oneTBB/archive/refs/tags/v2021.11.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 2552k    0 2552k    0     0  4600k      0 --:--:-- --:--:-- --:--:-- 4600k
Downloading https://github.com/Kitware/CMake/releases/download/v3.27.5/cmake-3.27.5-linux-x86_64.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 49.2M  100 49.2M    0     0   120M      0 --:--:-- --:--:-- --:--:--  120M
[termux_setup_cmake]: Applying cmake-c-compiler.patch
Downloading https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100  100k  100  100k    0     0   723k      0 --:--:-- --:--:-- --:--:--  723k
Archive:  /home/builder/.termux-build/libtbb/tmp/ninja-1.10.2.zip
  inflating: /home/builder/.termux-build/_cache/ninja-1.10.2/ninja  
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Android: Targeting API '24' with architecture 'arm64', ABI 'arm64-v8a', and processor 'aarch64'
-- The CXX compiler identification is Clang 17.0.2
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/builder/.termux-build/_cache/android-r26b-api-24-v0/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Looking for getcontext
-- Looking for getcontext - not found
-- The C compiler identification is Clang 17.0.2
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/builder/.termux-build/_cache/android-r26b-api-24-v0/bin/aarch64-linux-android-clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- HWLOC target HWLOC::hwloc_1_11 doesn't exist. The tbbbind target cannot be created
-- HWLOC target HWLOC::hwloc_2 doesn't exist. The tbbbind_2_0 target cannot be created
-- HWLOC target HWLOC::hwloc_2_5 doesn't exist. The tbbbind_2_5 target cannot be created
-- Configuring done (0.7s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_FIND_ROOT_PATH
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_USE_SYSTEM_LIBRARIES
    DOXYGEN_EXECUTABLE


-- Build files have been written to: /home/builder/.termux-build/libtbb/build
[1/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/allocator.cpp.o
[2/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/arena_slot.cpp.o
[3/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/address_waiter.cpp.o
[4/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/dynamic_link.cpp.o
[5/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/exception.cpp.o
[6/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/itt_notify.cpp.o
[7/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/concurrent_bounded_queue.cpp.o
[8/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/arena.cpp.o
[9/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/governor.cpp.o
[10/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/main.cpp.o
[11/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/market.cpp.o
[12/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/global_control.cpp.o
[13/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/misc.cpp.o
[14/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/misc_ex.cpp.o
[15/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/profiling.cpp.o
[16/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/tcm_adaptor.cpp.o
[17/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/rml_tbb.cpp.o
[18/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/parallel_pipeline.cpp.o
[19/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/observer_proxy.cpp.o
[20/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/private_server.cpp.o
[21/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/rtm_mutex.cpp.o
[22/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/semaphore.cpp.o
[23/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/version.cpp.o
[24/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/rtm_rw_mutex.cpp.o
[25/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/task.cpp.o
[26/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/small_object_pool.cpp.o
[27/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/task_group_context.cpp.o
[28/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/thread_dispatcher.cpp.o
[29/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/task_dispatcher.cpp.o
[30/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/__/tbb/itt_notify.cpp.o
[31/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/thread_request_serializer.cpp.o
[32/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/queuing_rw_mutex.cpp.o
[33/43] Building CXX object src/tbb/CMakeFiles/tbb.dir/threading_control.cpp.o
[34/43] Linking CXX shared library clang_17.0_cxx11_64_release/libtbb.so
[35/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backref.cpp.o
[36/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backend.cpp.o
[37/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/tbbmalloc.cpp.o
[38/43] Building CXX object src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/function_replacement.cpp.o
[39/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/large_objects.cpp.o
[40/43] Building CXX object src/tbbmalloc/CMakeFiles/tbbmalloc.dir/frontend.cpp.o
[41/43] Linking C shared library clang_17.0_cxx11_64_release/libtbbmalloc.so
FAILED: clang_17.0_cxx11_64_release/libtbbmalloc.so 
: && /home/builder/.termux-build/_cache/android-r26b-api-24-v0/bin/aarch64-linux-android-clang --target=aarch64-none-linux-android --gcc-toolchain=/home/builder/.termux-build/_cache/android-r26b-api-24-v0 --sysroot=/home/builder/.termux-build/_cache/android-r26b-api-24-v0/sysroot -fPIC -fstack-protector-strong -Oz --target=aarch64-linux-android24  -I/data/data/com.termux/files/usr/include -O3 -DNDEBUG  -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -fopenmp -static-openmp -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now --target=aarch64-linux-android24  -Wl,--version-script="/home/builder/.termux-build/libtbb/src/src/tbbmalloc/def/lin64-tbbmalloc.def"  -Wl,-z,relro,-z,now -shared -Wl,-soname,libtbbmalloc.so -o clang_17.0_cxx11_64_release/libtbbmalloc.so src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backend.cpp.o src/tbbmalloc/CMakeFiles/tbbmalloc.dir/backref.cpp.o src/tbbmalloc/CMakeFiles/tbbmalloc.dir/frontend.cpp.o src/tbbmalloc/CMakeFiles/tbbmalloc.dir/large_objects.cpp.o src/tbbmalloc/CMakeFiles/tbbmalloc.dir/tbbmalloc.cpp.o src/tbbmalloc/CMakeFiles/tbbmalloc.dir/__/tbb/itt_notify.cpp.o  -ldl  -pthread && :
ld.lld: error: version script assignment of 'local' to symbol 'ITT_DoOneTimeInitialization' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'TBB_runtime_interface_version' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'get_memcpy_largest_cachelinesize' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'get_memcpy_largest_cache_size' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'get_mem_ops_method' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'init_mem_ops_method' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'irc__get_msg' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'irc__print' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'override_mem_ops_method' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'set_memcpy_largest_cachelinesize' failed: symbol not defined
ld.lld: error: version script assignment of 'local' to symbol 'set_memcpy_largest_cache_size' failed: symbol not defined
clang-17: error: linker command failed with exit code 1 (use -v to see invocation)
[42/43] Building CXX object src/tbbmalloc_proxy/CMakeFiles/tbbmalloc_proxy.dir/proxy.cpp.o
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2023-11-17 18:19:44 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/6907487404">6907487404</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/libtbb/build.sh
+++ b/packages/libtbb/build.sh
@@ -2,12 +2,20 @@ TERMUX_PKG_HOMEPAGE=https://oneapi-src.github.io/oneTBB/
 TERMUX_PKG_DESCRIPTION="oneAPI Threading Building Blocks"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="@termux"
-TERMUX_PKG_VERSION="2021.10.0"
+TERMUX_PKG_VERSION="2021.11.0"
 TERMUX_PKG_SRCURL=https://github.com/oneapi-src/oneTBB/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=487023a955e5a3cc6d3a0d5f89179f9b6c0ae7222613a7185b0227ba0c83700b
+TERMUX_PKG_SHA256=782ce0cab62df9ea125cdea253a50534862b563f1d85d4cda7ad4e77550ac363
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_DEPENDS="libc++"
 TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
 -DTBB_STRICT=OFF
 -DTBB_TEST=OFF
 "
+
+termux_step_pre_configure() {
+	# Many symbols in the linker version scripts are undefined because link time
+	# optimization (-flto=thin) removes them. Suppress errors with lld >= 17 due to
+	# these undefined symbols.
+	# See https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=274337
+	LDFLAGS+=" -Wl,--undefined-version"
+}

